src/index.ts

```diff
@@ -1 +1,326 @@
+            if (!latestVersion) throw new Error(`No installable package found for: ${tool}`);
+
+            await installPackages(`${packageName};${latestVersion}`);
+            return getAndroidDevToolPath(tool);
+        }
+
+        throw new Error(`Unsupported tool: "${tool}"`);
+    }
+
+    const { tool: toolName, packageVersion } = tool;
+    if (!(toolName in versionedAndroidTools)) throw new Error(`Unsupported: ${toolName}`);
+
+    const { package: packageName } = versionedAndroidTools[toolName];
+    await installPackages(`${packageName};${packageVersion}`);
+    return getAndroidDevToolPath(tool);
+};
+
+/**
+ * Get the path to an Android development tool's executable. If the tool is not installed yet, it will automatically be
+ * installed.
+ *
+ * If no version is specified for a versioned tool, the latest installed (if the tool was already installed) or
+ * available version (if it isn't installed yet) is used.
+ *
+ * @param tool The tool to get the path for, either as just its name or with an explicit version.
+ *
+ * @returns The path to the installed tool's executable.
+ */
+export const getAndroidDevToolPath = async (tool: AndroidTool): Promise<string> => {
+    const { androidHome } = await ensureSdkmanager();
+
+    if (typeof tool === 'string') {
+        if (tool in unversionedAndroidTools) {
+            const relativeToolPath = unversionedAndroidTools[tool as keyof typeof unversionedAndroidTools].path;
+            const toolPath = join(androidHome, relativeToolPath);
+            if (!(await fs.exists(toolPath))) return await installAndroidDevTool(tool);
+            return toolPath;
+        } else if (tool in versionedAndroidTools) {
+            const relativeGlobToolPath = versionedAndroidTools[tool as keyof typeof versionedAndroidTools].path('*');
+            const relativeRegexToolPath =
+                versionedAndroidTools[tool as keyof typeof versionedAndroidTools].path('(.+)');
+
+            const installedTools = await globby(relativeGlobToolPath, { cwd: androidHome });
+            const installedVersions = installedTools
+                .map((p) => p.match(relativeRegexToolPath)?.[1])
+                .filter((v): v is string => !!v);
+            const latestVersion = getLatestVersion(installedVersions);
+
+            if (!latestVersion) return await installAndroidDevTool(tool);
+
+            return join(
+                androidHome,
+                versionedAndroidTools[tool as keyof typeof versionedAndroidTools].path(latestVersion)
+            );
+        }
+
+        throw new Error(`Unsupported tool: "${tool}"`);
+    }
+
+    const { tool: toolName, packageVersion } = tool;
+    if (!(toolName in versionedAndroidTools)) throw new Error(`Unsupported: ${toolName}`);
+
+    const relativeToolPath = versionedAndroidTools[toolName].path(packageVersion);
+    const toolPath = join(androidHome, relativeToolPath);
+
+    if (!(await fs.exists(toolPath))) return await installAndroidDevTool(tool);
```

Comment:
I see potential for an infinite loop here, if the version is wrong and nothing exits with an error during the install.